### ISSUE TITLE
Support @TestPropertySource as a repeated annotation

### ISSUE THREAD

**Issue by gebezs** [spring-projects/spring-framework#23299]
`TestPropertySource` is inherited in case of class hierarchy however it is not inherited in meta-annotation hierarchy.
The lack of this is made clear by @sbrannen in his answer: https://stackoverflow.com/a/26183692
He says:
> The algorithm that the Spring Framework uses when searching for an annotation stops once it has found the first occurrence of the sought annotation. 

## Expected behaviour
The settings from the `@CustomTest` and `MyTest` should merged as it is merged in case of class inheritance.

```java
@TestPropertySource(properties = { "property-a=value-a", "property-b=value-b" })
public @interface CustomTest {
}
```

```java
@CustomTest
@TestPropertySource(properties = { "property-a=value-x", "property-c=value-c" })
public class MyTest {
  @Value("${property-a}")
  private String propertyA;
  @Value("${property-b}")
  private String propertyB;
  @Value("${property-c}")
  private String propertyC;

  @Test
  public void test() {
    assertEquals("value-x", propertyA);
    assertEquals("value-b", propertyB);
    assertEquals("value-c", propertyC);
  }
}
```

## Rationale
It could give much more flexibility in tests.

**Comment by antkorwin** [comment]
This is same as #21986 and already fixed here #23320

**Comment by sbrannen** [comment]
> This is same as #21986 and already fixed here #23320

Indeed. Thanks for pointing that out.

For the sake of completeness, I tested the following against `master` (i.e., the upcoming Spring Framework 5.2), and it passes.

```java
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@TestPropertySource(properties = { "property-a=value-a", "property-b=value-b" })
@interface CustomTest {
}
```

```java
@RunWith(SpringRunner.class)
@ContextConfiguration
@TestPropertySource(properties = { "property-a=value-x", "property-c=value-c" })
@CustomTest
public class MyTest {

	@Value("${property-a}")
	private String propertyA;

	@Value("${property-b}")
	private String propertyB;

	@Value("${property-c}")
	private String propertyC;

	@Test
	public void test() {
		assertEquals("value-x", propertyA);
		assertEquals("value-b", propertyB);
		assertEquals("value-c", propertyC);
	}

	@Configuration
	static class Config {}
}
```

I am therefore closing this issue as a duplicate of gh-23320.

**Comment by gebezs** [comment]
Great news. @antkorwin, thanks for fixing this.
